tests/e2e/per_site_exporter_test.go

```diff
@@ -0,0 +1,357 @@
+				resp.Body.Close()
+				time.Sleep(100 * time.Millisecond)
+			}
+
+			time.Sleep(10 * time.Second)
+
+			metricsURL := fmt.Sprintf("https://%s.%s.svc.cluster.local:9302/metrics", serviceName, namespace)
+			c := newHTTPSClient(10 * time.Second)
+
+			Eventually(func() error {
+				resp, err := c.Get(metricsURL)
+				if err != nil {
+					return err
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("expected status 200, got %d", resp.StatusCode)
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					return err
+				}
+				metricsContent := string(body)
+				expected := []string{
+					"squid_site_requests_total",
+					"squid_site_misses_total",
+					"squid_site_bytes_total",
+					"squid_site_hit_ratio",
+					"squid_site_response_time_seconds",
+				}
+				for _, m := range expected {
+					if !strings.Contains(metricsContent, m) {
+						return fmt.Errorf("expected metric %s not found", m)
+					}
+				}
+				return nil
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("should have health check endpoint working", func() {
+			healthURL := fmt.Sprintf("https://%s.%s.svc.cluster.local:9302/health", serviceName, namespace)
+			c := newHTTPSClient(10 * time.Second)
+
+			Eventually(func() error {
+				resp, err := c.Get(healthURL)
+				if err != nil {
+					return err
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					return fmt.Errorf("expected status 200, got %d", resp.StatusCode)
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					return err
+				}
+				if string(body) != "OK" {
+					return fmt.Errorf("expected 'OK', got '%s'", string(body))
+				}
+				return nil
+			}, timeout, interval).Should(Succeed())
+		})
+	})
+
+	Describe("Per-Site Metrics with Traffic", func() {
+		var (
+			testServer    *testhelpers.ProxyTestServer
+			client        *http.Client
+			metricsClient *http.Client
+		)
+
+		BeforeEach(func() {
+			podIP, err := getPodIP()
+			Expect(err).NotTo(HaveOccurred(), "Failed to get pod IP")
+
+			var err2 error
+			testServer, err2 = testhelpers.NewProxyTestServer("Per-site exporter test", podIP, 0)
+			Expect(err2).NotTo(HaveOccurred(), "Failed to create test server")
+
+			client, err2 = testhelpers.NewSquidProxyClient(serviceName, namespace)
+			Expect(err2).NotTo(HaveOccurred(), "Failed to create proxy client")
+
+			metricsClient = newHTTPSClient(10 * time.Second)
+		})
+
+		AfterEach(func() {
+			if testServer != nil {
+				testServer.Close()
+			}
+		})
+
+		getPerSiteMetricsValue := func(metricsContent, metricName, hostname string) (float64, error) {
+			lines := strings.Split(metricsContent, "\n")
+			pattern := fmt.Sprintf(`^%s\{.*hostname="%s".*\}\s+([0-9.]+)`, regexp.QuoteMeta(metricName), regexp.QuoteMeta(hostname))
+			re := regexp.MustCompile(pattern)
+			for _, line := range lines {
+				if matches := re.FindStringSubmatch(line); len(matches) >= 2 {
```

Comment:
Can you add a comment here for how the check is working? I.e. what it's looking for and how it verifies that.